src/report.rs

```diff
@@ -336,41 +433,103 @@ function closeImageDialog() {
 document.getElementById('imageDialog').addEventListener('click', function(event) {
     closeImageDialog();
 });
+
+var selected = new Set();
+function toggle(event) {
+    let node = event.target.parentNode.parentNode;
+    let name = node.childNodes[2].textContent;
+    if (event.target.checked) {
+        selected.add(name);
+        node.style.color = \"#3a3\";
+    } else {
+        selected.delete(name);
+        node.style.color = \"#333\";
+    }
+    updateAcceptButton()
+}
+
+function updateAcceptButton() {
+    let text = document.getElementById('acceptText');
+    text.textContent = \"Accept selected cases (\" + selected.size + \" / \" + nTests + \")\";
+    let button = document.getElementById('acceptButton');
+    button.disabled = (selected.size === 0);
+}
+
+async function acceptTests() {
+    let text = document.getElementById('acceptText');
+    text.textContent = \"Updating \" + selected.size + \" cases ...\";
+    let button = document.getElementById('acceptButton');
+    button.disabled = true;
+
+    try {
+        const url = '/update';
+        const response = await fetch(url, {
+            method: 'POST',
+            headers: {
+               \"Content-Type\": \"application/json\",
+            },
+            body: JSON.stringify({ accepted_names: Array.from(selected) })
+        });
+        if (!response.ok) {
+          throw new Error(`Response status: ${response.status}`);
+        } else {
+          for (i = 0; i < nTests; i++) {
+                document.getElementById(\"t\" + i).checked = false;
+          }
+          location.reload();
```

Comment:
This is re-implementing the core logic of a HTML form. There's no advantage to this being a long-running file, as there aren't live updates. It might be better to just make this be an actual HTML form with checkboxes.